source/FilesServer/app/service.py

```diff
@@ -1,27 +1,25 @@
-from fastapi import HTTPException, status, Response
-from urllib.parse import quote
+from fastapi import HTTPException, status
+from fastapi.responses import RedirectResponse
 import boto3
 
 from app.settings import get_settings
 
 
-def get_bucket():  # TODO - add type hint
+def get_s3_client() -> boto3.client:
     settings = get_settings()
-    return boto3.resource("s3").Bucket(settings.aws_bucket_name)
+    return boto3.client("s3")
 
 
-async def s3_download(file_name: str) -> Response:
+async def s3_download(file_name: str) -> RedirectResponse:
     if not file_name:
         raise HTTPException(
             status_code=status.HTTP_400_BAD_REQUEST, detail="No file name provided"
         )
-    safe_file_name = quote(file_name)
-    bucket = get_bucket()
-    contents = bucket.Object(key=file_name).get()["Body"].read()
-    return Response(
-        content=contents,
-        headers={
-            "Content-Disposition": f"inline;filename={safe_file_name}",
-            "Content-Type": "application/octet-stream",
-        },
-    )
+
+    s3_client = get_s3_client()
+    presigned_url = s3_client.generate_presigned_url(
+        "get_object",
+        Params={"Bucket": "academease", "Key": file_name},
```

Comment:
i think those values (both the bucket name an the expires in should be in the app settings